spec.html

```diff
@@ -15552,18 +15574,25 @@ <h1>Runtime Semantics: PropertyDefinitionEvaluation</h1>
         <emu-alg>
           1. Let _propKey_ be the result of evaluating |PropertyName|.
           1. ReturnIfAbrupt(_propKey_).
-          1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) is *true*, then
+          1. If this |PropertyDefinition| is contained within a |Script| that is being evaluated for JSON.parse (see step <emu-xref href="#step-json-parse-eval"></emu-xref> of <emu-xref href="#sec-json.parse">JSON.parse</emu-xref>), then
+            1. Let _isProtoSetter_ be *false*.
+          1. Else if _propKey_ is the String value *"__proto__"* and if IsComputedPropertyKey of |PropertyName| is *false*, then
+            1. Let _isProtoSetter_ be *true*.
+          1. Else,
+            1. Let _isProtoSetter_ be *false*.
+          1. If IsAnonymousFunctionDefinition(|AssignmentExpression|) is *true* and _isProtoSetter_ is *false*, then
             1. Let _propValue_ be ? NamedEvaluation of |AssignmentExpression| with argument _propKey_.
           1. Else,
             1. Let _exprValueRef_ be the result of evaluating |AssignmentExpression|.
             1. Let _propValue_ be ? GetValue(_exprValueRef_).
+          1. If _isProtoSetter_ is *true*, then
+            1. If Type(_propValue_) is either Object or Null, then
+              1. Return ! _object_.[[SetPrototypeOf]](_propValue_).
```

Comment:
is it possible that this step invokes a Proxy trap?

Reply:
I don't believe so. As far as I'm aware the only thing which invokes this algorithm is the Evaluation semantics for object literals, and the object passed for `_object_` is provided by `OrdinaryObjectCreate(%Object.prototype%)`, which does not have any unusual behavior for `[[SetPrototypeOf]]`.